cms/djangoapps/contentstore/views/entrance_exam.py

```diff
@@ -0,0 +1,206 @@
+            if ee_min_score != '' and ee_min_score is not None:
+                entrance_exam_minimum_score_pct = float(ee_min_score)
+            return create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct)
+        return HttpResponse(status=400)
+
+    # Remove the entrance exam module for the specified course (returns 204 regardless of existence)
+    elif request.method == 'DELETE':
+        return delete_entrance_exam(request, course_key)
+
+    # No other HTTP verbs/methods are supported at this time
+    else:
+        return HttpResponse(status=405)
+
+
+def create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct):
+    """
+    api method to create an entrance exam.
+    First clean out any old entrance exams.
+    """
+    _delete_entrance_exam(request, course_key)
+    return _create_entrance_exam(request=request, course_key=course_key, entrance_exam_minimum_score_pct=entrance_exam_minimum_score_pct)
+
+
+def _create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct=50.0):
+    """
+    Internal workflow operation to create an entrance exam
+    """
+
+    # Confirm the course exists
+    course = modulestore().get_course(course_key)
+    if course is None:
+        return HttpResponse(status=400)
+
+    # Create the entrance exam item (currently it's just a chapter)
+    payload = {
+        'category': "chapter",
+        'display_name': "Entrance Exam",
+        'parent_locator': unicode(course.location),
+        'is_entrance_exam': True,
+    }
+    factory = RequestFactory()
+    internal_request = factory.post('/', json.dumps(payload), content_type="application/json")
+    internal_request.user = request.user
+    created_item = json.loads(create_item(internal_request).content)
+
+    # Set the entrance exam metadata flags for this course
+    # Reload the course so we don't overwrite the new child reference
+    course = modulestore().get_course(course_key)
+    metadata = {
+        'entrance_exam_enabled': True,
+        'entrance_exam_minimum_score_pct': entrance_exam_minimum_score_pct / 100,
+        'entrance_exam_id': created_item['locator'],
+    }
+    CourseMetadata.update_from_dict(metadata, course, request.user)
+
+    # Add an entrance exam milestone if one does not already exist
+    namespace = '{}.entrance_exams'.format(unicode(course_key))
+    milestones = milestones_api.get_milestones(namespace)
+    if len(milestones):
+        milestone = milestones[0]
+    else:
+        milestone = milestones_api.add_milestone({
+            'name': 'Completed Entrance Exam'.format(uuid4()),
+            'namespace': '{}.entrance_exams'.format(unicode(course_key)),
+            'description': 'Autogenerated during {} entrance exam creation workflow.'.format(
+                unicode(course.id)
+            ),
+        })
+    milestones_api.add_course_milestone(
+        unicode(course.id),
+        'requires',
```

Comment:
As I was mentioning in the edx-milestones reviews, I think it would be good to have these 'system defined' types as constants in the edx-milestones app so we can just refer to the constant here.